software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcess.java

```diff
@@ -147,6 +147,8 @@
     @SetFromFlag("runDir")
     AttributeSensorAndConfigKey<String,String> RUN_DIR = BrooklynConfigKeys.RUN_DIR;
 
+    ConfigKey<Boolean> ADD_OPEN_PORTS_EFFECTOR = ConfigKeys.newBooleanConfigKey("effector.add.openPorts", "Flag which adds effector for opening ports through Cloud security groups");
```

Comment:
I prefer giving an explicit default, rather than null.

(But I do see you handle null in SoftwareProcessImpl.init, so the code does work).